File: app/src/main/java/gtoken/com/nep2/object/O3Wallet.java
package gtoken.com.nep2.object;

import neoutils.Wallet;

import static neoutils.Neoutils.generatePublicKeyFromPrivateKey;

/**
 * Created by furyvn on 3/13/18.
 */

public class O3Wallet {

    private Wallet mO3Wallet;
    private String WIF;
    private String address;

    public String getWIF() {
        return WIF;
    }

    public void setWIF(String WIF) {
        this.WIF = WIF;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public O3Wallet(String rawPrivateKey) {
        try {
            mO3Wallet = generatePublicKeyFromPrivateKey(rawPrivateKey);
            this.WIF = mO3Wallet.getWIF();
            this.address = mO3Wallet.getAddress();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/gcox/wallet/object/O3Wallet.kt
package com.gcox.wallet.`object`

import neoutils.Wallet

import neoutils.Neoutils.generatePublicKeyFromPrivateKey

/**
 * Created by furyvn on 3/13/18.
 */

class O3Wallet(rawPrivateKey: String) {

    private var mO3Wallet: Wallet? = null
    lateinit var wif: String
    lateinit var address: String

    init {
        try {
            mO3Wallet = generatePublicKeyFromPrivateKey(rawPrivateKey)
            this.wif = mO3Wallet!!.wif
            this.address = mO3Wallet!!.address
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }
}
